File: test_project/gen_code_java/tbshape.java
package demo;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class TbShape {
    private final Map<int, Shape> _dataMap;
    private final List<Shape> _dataList;

    public TbShape(List<Shape> dataList) {
        this._dataList = dataList;
        this._dataMap = new HashMap<>();
        for (Shape v : _dataList) {
            this._dataMap.put(v.Id, v);
        }
    }

    public Map<int, Shape> getDataMap() {
        return _dataMap;
    }

    public List<Shape> getDataList() {
        return _dataList;
    }

    public Shape get(int key) {
        return _dataMap.get(key);
    }
}

File: test_project/gen_code_lua/tbshape.lua
-- Table TbShape
local TbShape = {
    _dataMap = {},
    _dataList = {}
}

function TbShape.Load(dataList)
    TbShape._dataList = dataList
    for _, v in ipairs(dataList) do
        TbShape._dataMap[v.Id] = v
    end
end

function TbShape.Get(key)
    return TbShape._dataMap[key]
end

function TbShape.GetDataList()
    return TbShape._dataList
end

function TbShape.GetDataMap()
    return TbShape._dataMap
end

return TbShape